lib/services/history/sources/remote_history.ts

```diff
@@ -0,0 +1,71 @@
+import { SNApiService } from "@Lib/services/api/api_service";
+import { SNProtocolService } from "@Lib/services/protocol_service";
+import { ItemHistory } from "../item_history";
+import { HttpResponse } from "@Lib/services/api/http_service";
+import { RawPayload, CreateMaxPayloadFromAnyObject, CreateSourcedPayloadFromObject } from "@Lib/protocol/payloads/generator";
+import { ItemHistorySource, ItemHistoryEntry } from "../item_history_entry";
+import { PayloadSource } from "@Lib/protocol/payloads";
+
+export class RemoteHistory {
+  private apiService: SNApiService
+  private protocolService: SNProtocolService
+
+  constructor(apiService: SNApiService, protocolService: SNProtocolService) {
+    this.apiService = apiService;
+    this.protocolService = protocolService;
+  }
+
+  /**
+   * Iterates over the response and creates a payload from each entry.
+   * @param response
+   */
+  private responseToPayloadArray(response: HttpResponse) {
+    let revisions: RawPayload[] = response!.revisions;
+    const revisionEntries = revisions.map((revision) => {
+      return {
+        payload: CreateMaxPayloadFromAnyObject(revision)
+      };
+    });
+    return {
+      entries: revisionEntries
+    };
+  }
+
+  public async fetchItemHistory(itemUuid: string) {
+    const serverResponse = await this.apiService!.getItemRevisions(itemUuid);
+    if (serverResponse.error) {
+      return {
+        entries: []
+      };
+    }
+    const payloadArray = this.responseToPayloadArray(serverResponse);
+    const itemHistory = ItemHistory.FromJson(payloadArray, ItemHistorySource.Remote);
+    return itemHistory;
+  }
+
+  /**
+   * Creates a PurePayload from the response.
+   * @param response
+   */
+  private responseToPayload(response: HttpResponse) {
+    delete response.error;
+    delete response.status;
```

Comment:
Should definitely not be mutating input response object, especially unexpectedly because you don't know the caller doesn't require these properties.

These properties should be ignored by `CreateMaxPayloadFromAnyObject` anyway since they're not a `PayloadField`.

You can also try using something like `response as Partial<RawPayload>` instead of `unknown`.